packages/vite-plugin-angular/src/lib/angular-vite-plugin.ts

```diff
@@ -244,8 +257,19 @@ export function angular(
             >('@angular/compiler-cli/linker/babel')
           ).createEs2015LinkerPlugin;
 
+          const forceAsyncTransformation =
```

Comment:
The check for the optimizer should be closer to these lines, along with the returned object of `code`

https://github.com/angular/angular-cli/pull/23682/files#diff-309812edd0ff24c480525d73b2e4b7a0d82b91d2a689720f2e77395f7958bea7R371

Reply:
Do you mean closer as in it should look more like the code referenced here?

Reply:
The code I referenced should be copied into the transform for the optimizer plugin section. It's not the same as the code in the vite-plugin transform section

Reply:
Okay. Sorry for all the questions 🙈

I think this is more what you're looking for:

```
const forceAsyncTransformation =
  !/[\\/][_f]?esm2015[\\/]/.test(id) &&
  /for\s+await\s*\(|async\s+function\s*\*/.test(code);
const shouldLink = await requiresLinking(id, code);
const useInputSourcemap =
  pluginOptions.sourcemap &&
  (!!pluginOptions.thirdPartySourcemaps || !/[\\/]node_modules[\\/]/.test(id));

// If no additional transformations are needed, return the TypeScript output directly
if (!forceAsyncTransformation && !isProd && !shouldLink) {
  return {
    // Strip sourcemaps if they should not be used
    code: useInputSourcemap
      ? code
      : code.replace(/^\/\/# sourceMappingURL=[^\r\n]*/gm, ""),
  };
}

if (!forceAsyncTransformation && !isProd) {
  return {
    code: !isProd
      ? code
      : code.replace(/^\/\/# sourceMappingURL=[^\r\n]*/gm, ""),
  };
}
```

The linked code you posted has reference to `pluginOptions.sourcemap` and `pluginOptions.thirdPartySourcemaps`. I'm not sure what the equivalent would be here. In addition, the two returns in the if statements here are the same, so I'm not sure which I should use, or if both should be present but one changed at all.

Reply:
No problem! The shouldLink is present in the second transform but not in the first. I think that's the biggest difference. The isProd should take the place of any changes for advanced optimizations and sourcemaps flags.